versioncontrol/gerrit_test.go

```diff
@@ -135,13 +137,18 @@ func TestReportObjFromGerrit(t *testing.T) {
 					Version: "",
 				},
 			},
+			wantErr: errors.New("unable to retrieve license for golang.org/x/lint"),
 		},
 	}
 	for _, test := range tests {
 		t.Run(test.description, func(t *testing.T) {
 			reportObject, err := ReportObjFromGerrit(test.dependency, request)
 			if err != nil {
-				t.Errorf("unable to compile report for gerrit: %v", err)
+				if test.wantErr != nil {
+					assert.EqualError(t, err, test.wantErr.Error())
```

Comment:
So you are wanting an error, construct an error to compare against, and then check the message? 

What about something like:

```go
var lintError error := errors.New("unable to retrieve license for golang.org/x/lint")
...
wantErr: lintError
...
if test.wantErr != nil {
    assertEqual(t, err, lintError)
}
```

Ideally `lintError` would have already been defined elsewhere in the repo and you are throwing a known error type.

Reply:
Yes, that's the idea.

The concern that I have about this approach is that we would have to define a specific error for each possible dependency. I'm not sure that adds value in this particular case.